packages/backend/src/helpers/authentication.ts

```diff
@@ -13,12 +13,25 @@ export const setCurrentUserContext = async ({
   req: Request
   res: Response
 }): Promise<UnauthenticatedContext> => {
-  const context: UnauthenticatedContext = { req, res, currentUser: null }
+  const context: UnauthenticatedContext = {
+    req,
+    res,
+    currentUser: null,
+    isAdminQuery: false,
+  }
 
   // Get tiles view key from headers
   context.tilesViewKey = req.headers['x-tiles-view-key'] as string | undefined
 
-  context.currentUser = await getLoggedInUser(req)
+  if (typeof req.headers['x-plumber-admin-token'] === 'string') {
+    context.currentUser = await validateAdminTokenAndGetUser(
+      req.headers['x-plumber-admin-token'],
+    )
+    context.isAdminQuery = true
```

Comment:
if u want to include permission level, this key name might need to reflect that.

Reply:
renamed to `isAdminOperation`